corehq/apps/domain/decorators.py

```diff
@@ -315,11 +347,13 @@ def mobile_auth_or_formplayer(view_func):
 login_or_digest = login_or_digest_ex()
 login_or_basic = login_or_basic_ex()
 login_or_api_key = login_or_api_key_ex()
+login_or_oauth2 = login_or_oauth2_ex()
 
 # Use these decorators on views to exclusively allow any one authorization method and not session based auth
 digest_auth = login_or_digest_ex(allow_sessions=False)
 basic_auth = login_or_basic_ex(allow_sessions=False)
 api_key_auth = login_or_api_key_ex(allow_sessions=False)
+oauth2_auth = login_or_oauth2_ex(allow_sessions=False)
```

Comment:
Man our auth decorators are messy and weird.

Reply:
agreed. i hate how all this works but was afraid to touch it.

Reply:
wow, it is way more difficult than I expected to make an API that checks that a user can authenticate but not check anything about domains.